### PR TITLE
Add HarvestDropsEvent

### PR DESCRIPTION
**Pull request by cout970**
This RP add the ability to modify the items that blocks drops

### CODE REVIEW

File: src/main/java/xyz/openmodloader/event/impl/BlockEvent.java [hunk 1]
```diff
@@ -157,6 +163,113 @@ public static float onDig(float digSpeed, World world, IBlockState state, BlockP
...
+         * @param world The world where the event took place.
+         * @param state The state of the block involved with the event.
+         * @param pos The position of the event.
+         * @param chance The probability to drop one item
+         * @param fortune The level of fortune used to mine the block
+         * @param initialDrops The list of items that the block can drop without applying chance
+         * @param finalDrops The list of items that the block will drop
+         */
+        public HarvestDrops(World world, IBlockState state, BlockPos pos, float chance, int fortune,
+                            ImmutableList<ItemStack> initialDrops, List<ItemStack> finalDrops) {
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/BlockEvent.java]: Most things shouldn't be line wrapped.

File: src/main/java/xyz/openmodloader/event/impl/BlockEvent.java [hunk 2]
```diff
@@ -157,6 +163,113 @@ public static float onDig(float digSpeed, World world, IBlockState state, BlockP
...
+         * @param state The state of the block involved with the event.
+         * @param pos The position of the event.
+         * @param chance The probability to drop one item
+         * @param fortune The level of fortune in the tool used to mine the block
+         * @param initialDrops The list of drops that the block can drop
+         * @return The list of items that the block will drop
+         */
+        public static List<ItemStack> onHarvestDrops(World world, IBlockState state, BlockPos pos, float chance, int fortune, List<ItemStack> initialDrops){
+            List<ItemStack> finalDrops = initialDrops.stream().filter(stack -> world.rand.nextFloat() <= chance)
+                    .map(ItemStack::copy).collect(Collectors.toCollection(LinkedList::new));
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/BlockEvent.java]: Also line wrapping issues here. Please use same line.

File: src/main/java/xyz/openmodloader/event/Events.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    public boolean post(T event) {
+        for (IEventExecutor<T> executor : this.executorList) {
+            executor.execute(event);
+            if (event.isCanceled()) {
+                return false;
+            }
+        }
+        return true;
+    }
+}
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/Events.java]: This class was removed in https://github.com/OpenModLoader/OpenModLoader/commit/0556c2401b109411f7665ae6e02ca6f19fbf31c2 no reason to add it back in.

File: src/main/java/xyz/openmodloader/test/OMLTestMod.java [hunk 4]
```diff
@@ -82,4 +84,10 @@ private void onScreenshot(ScreenshotEvent event) {
         graphics.setFont(new Font("Arial Black", Font.BOLD, 20));
         graphics.drawString("Open Mod Loader", 20, 40);
     }
+
+    private void onHarvestDrops(BlockEvent.HarvestDrops event){
+        OpenModLoader.INSTANCE.LOGGER.info("Dropping items: "+event.getFinalDrops()+" from original items: "
+                +event.getInitialDrops()+", with fortune: "+event.getFortune()+", with chance: "+event.getChance());
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/test/OMLTestMod.java]: And line wrapping issues here as well.

File: src/main/java/xyz/openmodloader/event/Events.java (hunk 3 above)
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/event/Events.java]: Yep, this PR needs to be updated to use the unified event bus.

**Comment by cout970** [comment]
BTW, I found a bug, https://github.com/OpenModLoader/OpenModLoader/blob/master/src/main/java/xyz/openmodloader/event/impl/BlockEvent.java#L155
this should be:
return !OpenModLoader.INSTANCE.EVENT_BUS.post(event) || event.getDigSpeed() < 0F ? 0f : event.getDigSpeed();

**Pull request merged**